DNN Platform/Website/admin/Skins/Logo.ascx.cs

```diff
@@ -58,6 +64,40 @@ protected override void OnLoad(EventArgs e)
                         {
                             this.imgLogo.ImageUrl = imageUrl;
                             logoVisible = true;
+                        }
+
+                        if (this.InjectSVG.GetValueOrDefault() == true)
+                        {
+                            logoVisible = false; // hide the <img>, we are going to inject an <svg> tag instead.
+
+                            hypLogo.CssClass = this.CssClass; // put the custom CSS class on the hyperlink instead
+
+                            string svg = string.Empty;
+                            string cacheSvg = (string)CachingProvider.Instance().GetItem(this._cacheKey);
+
+                            if (string.IsNullOrEmpty(cacheSvg))
+                            {
+                                if (fileInfo.Extension == "svg")
+                                {
+                                    XDocument svgXmlDoc = XDocument.Load(fileInfo.PhysicalPath);
+                                    XElement svgXmlNode = svgXmlDoc.Descendants().Where(x => x.Name.LocalName == "svg").SingleOrDefault();
+
+                                    if (svgXmlNode != null)
+                                    {
+                                        svg = svgXmlNode.ToString();
+                                        CachingProvider.Instance().Insert(this._cacheKey, svg);
+                                    }
+                                }
+                            }
+                            else
+                            {
+                                svg = cacheSvg;
+                            }
+
+                            Literal litSvg = new Literal();

```

Comment:
We need some sort of fallback at this point if the file couldn't be obtained.

Reply:
What would you suggest here?  Inject the Portal Name as string literal?

Reply:
I'm not sure on this, but there needs to be something somewhere I think

Reply:
Also, this still needs a try/catch around it